src/integrations/third-party/elementor.php

```diff
@@ -494,6 +494,28 @@ protected function render_hidden_fields() {
 		echo '</form>';
 	}
 
+	/**
+	 * Returns the slug for the post being edited.
+	 *
+	 * @return string
+	 */
+	protected function get_post_slug() {
+		$post = $this->get_metabox_post();
+
+		// In case get_metabox_post returns null for whatever reason.
+		if ( ! $post instanceof WP_Post ) {
+			return '';
+		}
+
+		// Drafts might not have a post_name unless the slug has been manually changed.
+		// In this case we get it using get_sample_permalink.
+		if ( ! $post->post_name ) {
+			return \get_sample_permalink( $post->id )[1];
```

Comment:
One concern: the `get_sample_permalink` method applies filters, so might be used by third parties. See line 1436 of the source code at https://developer.wordpress.org/reference/functions/get_sample_permalink/.

I'm afraid this is a "better safe than sorry" request again, that we should check whether an array is returned (and has a count of 2?).

Reply:
Done! Also check that the value in the array is a string.